packages/backend/src/app/instance/instance.service.ts

```diff
@@ -78,6 +81,29 @@ export const instanceService = {
             projectId
         });
     },
+    async updateInstanceStatusByCollectionId({ projectId,request }: { projectId: ProjectId, request: UpsertInstanceRequest }) {
+        const collection = await collectionService.getOne({ projectId: projectId, id: request.collectionId});
+        if (collection == null) {
+            throw new ActivepiecesError({
+                code: ErrorCode.COLLECTION_NOT_FOUND,
+                params: {
+                    id: request.collectionId,
+                },
+            });
+        }
+        const instance: Instance | null = await instanceRepo.findOneBy({ projectId, collectionId: request.collectionId });
+        if(instance) {
+            instance.status = request.status;
```

Comment:
You also need to handle that the state where old state is not equal to new state.